packages/language/src/workspace/compilation-unit.ts

```diff
@@ -132,10 +143,20 @@ const getRootPreprocessorScope = createBuiltinScopeGetter(
 export async function createCompilationUnit(
   uri: URI,
 ): Promise<CompilationUnit> {
+  const compilerOptions = getDefaultCompilerOptions();
+  const services: CompilationServices = {
+    files: new FileStore(),
+    typeCache: new DefaultTypeCache(),
+    inferer: new DefaultTypeInferer(
+      (compilerOptions.rules?.ibm ?? "ANS") === "ANS"
+        ? CompilerOptionRules.ANS
+        : CompilerOptionRules.IBM,
+    ),
```

Comment:
Suggestion: I don't think it's a good idea to provide the value as a constructor argument to the service - AFAIK the `IBM` compiler option can be changed during the lifetime of the `CompilationUnit`.